platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/Capture.kt

```diff
@@ -512,14 +704,17 @@ object Capture {
          */
         @JvmStatic
         fun setSleepMode(sleepMode: SleepMode) {
-            logger()?.setSleepMode(sleepMode)
+            getInternalLogger()?.setSleepMode(sleepMode)
         }
 
         /**
          * Used for testing purposes.
          */
+        @VisibleForTesting
         internal fun resetShared() {
             default.set(LoggerState.NotStarted)
         }
+
+        private fun getInternalLogger(): ILogger? = preInitInMemoryLoggerRef.get() ?: logger()
```

Comment:
This feels like it should just be a single reference that we change over once the real logger is ready.